src/cxx.cc

```diff
@@ -783,15 +784,150 @@ static_assert(sizeof(std::string) <= kMaxExpectedWordsInString * sizeof(void *),
+  template <>                                                                  \
+  void Option<CXX_TYPE const *>::drop() noexcept {                             \
+    cxxbridge1$rust_option$const$##RUST_TYPE##$drop(this);                     \
+  }                                                                            \
+  template <>                                                                  \
+  bool Option<CXX_TYPE const *>::has_value() const noexcept {                  \
+    return cxxbridge1$rust_option$const$##RUST_TYPE##$has_value(this);         \
+  }                                                                            \
+  template <>                                                                  \
+  CXX_TYPE const ** Option<CXX_TYPE const *>::value_ptr() noexcept {           \
+    return cxxbridge1$rust_option$const$##RUST_TYPE##$value_ptr(this);         \
+  }                                                                            \
+  template <>                                                                  \
+  void Option<CXX_TYPE const *>::set(CXX_TYPE const *&& value) noexcept {      \
+    return cxxbridge1$rust_option$const$##RUST_TYPE##$set(                     \
+      this, std::move(value));                                                 \
+  }                                                                            \
+  template <>                                                                  \
+  Option<CXX_TYPE *>::Option() noexcept {                                      \
+    cxxbridge1$rust_option$##RUST_TYPE##$new(this);                            \
+  }                                                                            \
+  template <>                                                                  \
+  void Option<CXX_TYPE *>::drop() noexcept {                                   \
+    cxxbridge1$rust_option$##RUST_TYPE##$drop(this);                           \
+  }                                                                            \
+  template <>                                                                  \
+  bool Option<CXX_TYPE *>::has_value() const noexcept {                        \
+    return cxxbridge1$rust_option$##RUST_TYPE##$has_value(this);               \
+  }                                                                            \
+  template <>                                                                  \
+  CXX_TYPE**  Option<CXX_TYPE *>::value_ptr() noexcept {                       \
+    return cxxbridge1$rust_option$##RUST_TYPE##$value_ptr(this);               \
+  }                                                                            \
+  template <>                                                                  \
+  void Option<CXX_TYPE *>::set(CXX_TYPE *&& value) noexcept {                  \
+    return cxxbridge1$rust_option$##RUST_TYPE##$set(this, std::move(value));   \
+  }    \
```

Comment:
I was a bit surprised to see `Option<CXX_TYPE *>` here. Do we support `rust::Option<int32_t>` for instance? Or would it have to instead be `rust::Option<int32_t *>`? It feels a bit odd to see the latter coming from other c++ codebases

Reply:
No only pointers because we rely on the niche optimization :( I will try to support Option<T> for builtin types like ints and bools in a follow up but I have no idea if it's possible so no promises